rust/stackable-cockpit/src/platform/stacklet/mod.rs

```diff
@@ -113,10 +113,14 @@ async fn list_stackable_stacklets(
     client: &Client,
     namespace: Option<&str>,
 ) -> Result<Vec<Stacklet>, Error> {
-    let product_list = build_products_gvk_list(PRODUCT_NAMES);
     let mut stacklets = Vec::new();
 
-    for (product_name, product_gvk) in product_list {
+    for (product_name, group, version, kind) in PRODUCTS {
+        let product_gvk = GroupVersionKind {
+            group: group.to_string(),
+            version: version.to_string(),
+            kind: kind.to_string(),
+        };
```

Comment:
suggestion: Use `to_owned` here instead.


```suggestion
        let product_gvk = GroupVersionKind {
            group: group.to_owned(),
            version: version.to_owned(),
            kind: kind.to_owned(),
        };
```

Reply:
Unfortunately, the fields have the type `&&str`. Therefore, `to_owned()` does not work.

Some possible options are:

```rust
group: group.to_owned().to_owned()

group: (*group).to_owned()

#[allow(suspicious_double_ref_op)]
group: group.clone().to_owned()
```

Reply:
We decided to leave it as is, marking as resolved.